### PR TITLE
Refactor ScopeExp::semantic()

### PR DESCRIPTION
**Pull request by 9rnsr**
Separated from #4731.

Remove goto statement, and reduce indent level.

### CODE REVIEW

**Comment by 9rnsr** [comment]
@WalterBright Ping. This is required to improve detection for the variable circular reference/initialization.

File: src/expression.d [hunk 1]
```diff
@@ -5195,9 +5195,10 @@ public:
         }
         //if (type == Type::tvoid)
         //    return this;
-    Lagain:
+
+        ScopeDsymbol sds = this.sds;
```
Comment (yebblies) [inline src/expression.d]: It was better when this didn't shadow a member variable.

Comment (9rnsr) [inline src/expression.d]: OK, renamed to `sds2`.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
@yebblies Thanks!

**Comment by WalterBright** [comment]
yeah, looks good

**Pull request merged**